### DISCUSSION TITLE
Model::with() can't load relation data when foreign key include letter "e"

### DISCUSSION THREAD

**Discussion by FuS1** [laravel/framework#43472] [Q&A]
- Laravel Version: 8.83.3
- PHP Version: 7.4.3
- Database Driver & Version: MYSQL 5.6

### Description:

I have two table "archives" and "archives_permission" and data

![1659082487771](https://user-images.githubusercontent.com/5563956/181715545-1e6143b5-b9c8-47c8-b97d-440c4903b406.jpg)

![1659082556320](https://user-images.githubusercontent.com/5563956/181715780-1759801d-3ea4-4acd-be72-de7eab6f600b.jpeg)

and Models/Archives.php  

```
 public function archivesPermission()
  {
      return $this->hasMany(ArchivesPermission::class,'archives_uuid','uuid');
  }
```

When I use `Archives::with(['archivesPermission'])->first()` ,then archivesPermission is empty collection
![1659083092241](https://user-images.githubusercontent.com/5563956/181717339-c1f75729-bc4b-4300-b217-0d473a208741.jpg)

But When I use `Archives::first()->archivesPermission()->get()`  will get correct data
![image](https://user-images.githubusercontent.com/5563956/181717196-11f22e5d-72a8-4af6-925b-a22b54edb2eb.png)

If I change uuid from "96**e**45fe0-27d2-499e-8bbc-79c4e1d4d015" to "96**d**45fe0-27d2-499e-8bbc-79c4e1d4d015", then 
`Archives::with(['archivesPermission'])->first()` ,then archivesPermission will get correct data

**Comment by macropay-solutions** [comment]
define
 $keyType='string'; in your model
